src/UICrafter.Client/Components/AppViewDesigner.razor

```diff
@@ -0,0 +1,87 @@
+            </MudDropZone>
+        </ChildContent>
+        <ItemRenderer>
+            @if (context is UIButton button)
+            {
+                <MudButton Class="mb-2" Variant="Variant.Filled" @onfocus="()=>SelectComponent(button)">@button.Label</MudButton>
+            }
+            else if (context is UIInputField inputField)
+            {
+                <MudTextField @bind-Value="@inputField.Value" Class="mb-2" Label="@inputField.Placeholder" Variant="Variant.Outlined" @onfocus="()=>SelectComponent(inputField)"></MudTextField>
+            }
+            else if (context is UITextBox textBox)
+            {
+                <MudTextField @bind-Value="@textBox.Content" Label="@textBox.Label" ReadOnly="true" ShrinkLabel="true" Lines=@textBox.NumberOfLines Variant="Variant.Outlined" @onfocus="()=>SelectComponent(textBox)" />
+            }
+            else
+            {
+                <MudPaper Elevation="4" Class="pa-4 my-4">Unknown Component</MudPaper>
+            }
+        </ItemRenderer>
+    </MudDropContainer>
+</MudContainer>
+
+@code {
+    [Parameter]
+    public required List<UIComponent> UIComponents { get; set; }
+
+    private UIComponent? _internalSelectedComponent;
+    [Parameter]
+    public required UIComponent? SelectedComponent
+    {
+        get => _internalSelectedComponent;
+        set
+        {
+            if (SelectedComponentChanged.HasDelegate && value != _internalSelectedComponent)
+            {
+                _internalSelectedComponent = value;
+                SelectedComponentChanged.InvokeAsync(value);
+            }
+        }
+    }
+    [Parameter]
+    public required EventCallback<UIComponent?> SelectedComponentChanged { get; set; }
+
+    private MudDropContainer<UIComponent>? _internalDropContainer;
+    [Parameter]
+    public required MudDropContainer<UIComponent>? DropContainer
+    {
+        get => _internalDropContainer;
+        set
+        {
+            if(DropContainerChanged.HasDelegate && value != _internalDropContainer)
+            {
+                _internalDropContainer = value;
+                DropContainerChanged.InvokeAsync(value);
+            }
+        }
+    }
+    [Parameter]
+    public required EventCallback<MudDropContainer<UIComponent>?> DropContainerChanged { get; set; }
+
+    private void SelectComponent(UIComponent component) => SelectedComponent = component;
+
+    private void ComponentUpdated(MudItemDropInfo<UIComponent> component)
+    {
+        if (component is null || component.Item is null) return;
+
+        int oldIndex = UIComponents.IndexOf(component.Item);
+        int newIndex = component.IndexInZone;
+
+        if (oldIndex != newIndex)
+        {
+            var tmp = UIComponents[oldIndex];
+            UIComponents.RemoveAt(oldIndex);
+            UIComponents.Insert(newIndex, tmp);
+            // Console.WriteLine("Old: " + oldIndex + "\n" + "New: " + newIndex);
```

Comment:
If you want to log in dotnet you should use the ILogger setup with LogDebug(). There are a lot of good stuff with this but we need a bit more general setup to do it. I have created a task about it we should look into soon.

Reply:
Virkede fint nok med CW til at debug med

Reply:
Det er ikke så meget at det ikke vil fungere med CW. Men hvis du laver LogDebug så kan det ændres ved appsettings om det skal printes så man bare kan beholde loggings og så disable dem bare ved at ændre en setting. :)